pennylane_lightning/core/src/simulators/lightning_gpu/gates/tests/Test_StateVectorCudaManaged_NonParam.cpp

```diff
@@ -1105,4 +1105,250 @@ TEMPLATE_TEST_CASE("StateVectorCudaManaged::SetIthStates",
 
         CHECK(expected_state == Pennylane::Util::approx(sv.getDataVector()));
     }
-}
\ No newline at end of file
+}
+
+TEMPLATE_TEST_CASE("StateVectorCudaManaged::applyOperation non-param "
+                   "one-qubit with controls",
+                   "[StateVectorCudaManaged]", float, double) {
+    using PrecisionT = TestType;
+    std::mt19937 re{1337};
+    const int num_qubits = 4;
+    const auto margin = PrecisionT{1e-5};
+    const std::size_t control = GENERATE(0, 1, 2, 3);
+    const std::size_t wire = GENERATE(0, 1, 2, 3);
```

Comment:
Eeach `GENERATE` macro will generate one test for one entry. Take `const std::size_t control = GENERATE(0, 1, 2, 3);` as an example, it will generate 4 runs with control = 0, 1, 2,3; In the testing case here, with Generate, there will be 16 tests with different combinations of `control` and `wire`.